### PR TITLE
Features/timetable block

### PR DESCRIPTION
**Pull request by ToufiPF**
Adds Timetables to the app

Before:
<insert empty timetable here>

After:
Screenshot_1615396488
Screenshot_1615396499

### CODE REVIEW

**Review by lucasrlt — APPROVED** [review]
Everything looks fine to me, great PR !

File: app/src/main/java/com/pindex/main/ui/fragments/blocks/TimetableBlockFragment.kt [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            specialLayout.visibility = View.GONE
+        if (regularLayout.childCount == 0)
+            regularLayout.visibility = View.GONE
+    }
+
+
+    private fun appendAllEntriesToSpecialLayout(special: List<TimetableBlockDto.EntryDto>) {
+        for (e in special)
+            fillLayoutWithSpecialEntry(specialLayout, e)
+    }
```
Comment (lucasrlt) [inline app/src/main/java/com/pindex/main/ui/fragments/blocks/TimetableBlockFragment.kt:141]: There is a missing space, you could maybe run Code > Reformat Code in Android Studio on that file ?

File: app/src/main/java/com/pindex/main/ui/fragments/blocks/TimetableBlockFragment.kt [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                            timestampToString(start) + " au " + timestampToString(end)
+                    if (e.reason != null)
+                        msg += System.lineSeparator() + e.reason
+                    view.text = msg
+
+                    lay.addView(view)
+                }
+            }
+        }
+    }
```
Comment (lucasrlt) [inline app/src/main/java/com/pindex/main/ui/fragments/blocks/TimetableBlockFragment.kt:172]: Same, missing space

File: app/src/main/java/com/pindex/main/utils/Constants.kt [hunk 3]
```diff
@@ -50,6 +50,16 @@ object Constants {
          */
         const val TEXT_BLOCK_TEXT: String = "text"
 
+        /**
+         * The list of entries given as argument to a TimetableBlockFragment
+         */
+        const val TIMETABLE_REGULAR_ENTRY_LIST = "timetableEntries"
+        /**
```
Comment (lucasrlt) [inline app/src/main/java/com/pindex/main/utils/Constants.kt:57]: Missing space here, same comment as in TimetableBlockFragment

File: app/src/test/java/com/pindex/main/ui/fragments/TimetableBlockFragmentTest.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+class TimetableBlockFragmentTest {
+    @Test
+    fun timetableLayoutsAreCreated() {
+        val block = TimetableBlockFragment()
+
+        val regular = block.view?.findViewById<TableLayout>(R.id.pindex_block_timetable_regular_entries)
+        val special = block.view?.findViewById<LinearLayout>(R.id.pindex_block_timetable_special_entries)
+        // Note: regular & special have not been created:
+        //they are created in onViewCreated() which is not called in this situation
```
Comment (lucasrlt) [inline app/src/test/java/com/pindex/main/ui/fragments/TimetableBlockFragmentTest.kt:20]: Missing space between // and the comment

**Review by lucasrlt — CHANGES_REQUESTED** [review]
The code is fine, it would be good to run Android Studio's linter on some files for code consistency :)

**Comment by aouame** [comment]
Code looks in order for me. I went through missing spaces and everything seems fines from second commit.

**Review by aouame — APPROVED** [review]

**Review by lucasrlt — APPROVED** [review]
Looks good, the code is consistent

**Pull request merged**